nav2_route/src/corner_arc.cpp

```diff
@@ -0,0 +1,133 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_route/corner_arc.hpp"
+#include <iostream>
+
+namespace nav2_route
+{
+
+CornerArc::CornerArc(EdgePtr start_edge, EdgePtr end_edge, double minimum_radius)
+{
+  start_edge_ = start_edge;
+  end_edge_ = end_edge;
+
+  double angle = getAngleBetweenEdges(start_edge, end_edge);
+
+  double tangent_length = minimum_radius/(std::tan(angle/2));
+
+  start_edge_length_ = getEdgeLength(start_edge);
+  end_edge_length_ = getEdgeLength(end_edge);
+
+  if(tangent_length < start_edge_length_ && tangent_length < end_edge_length_){
+    std::vector<double> start_edge_unit_tangent;
+    std::vector<double> end_edge_unit_tangent;
+    std::vector<double> unit_bisector;
+
+    start_edge_unit_tangent.push_back((start_edge->start->coords.x -start_edge->end->coords.x)/start_edge_length_);
+    start_edge_unit_tangent.push_back((start_edge->start->coords.y -start_edge->end->coords.y)/start_edge_length_);
+    end_edge_unit_tangent.push_back((end_edge->end->coords.x - end_edge->start->coords.x)/end_edge_length_);
+    end_edge_unit_tangent.push_back((end_edge->end->coords.y - end_edge->start->coords.y)/end_edge_length_);
+
+    double bisector_x = start_edge_unit_tangent[0]+end_edge_unit_tangent[0];
+    double bisector_y = start_edge_unit_tangent[1]+end_edge_unit_tangent[1];
+    double bisector_magnitude = std::sqrt(bisector_x*bisector_x + bisector_y*bisector_y);
+
+    unit_bisector.push_back(bisector_x/bisector_magnitude);
+    unit_bisector.push_back(bisector_y/bisector_magnitude);
+
+    start_coordinate_.x = start_edge->end->coords.x + start_edge_unit_tangent[0]*tangent_length;
+    start_coordinate_.y = start_edge->end->coords.y + start_edge_unit_tangent[1]*tangent_length;
+
+    end_coordinate_.x = end_edge->start->coords.x + end_edge_unit_tangent[0]*tangent_length;
+    end_coordinate_.y = end_edge->start->coords.y + end_edge_unit_tangent[1]*tangent_length;
+
+    double signed_angle = getSignedAngleBetweenEdges(start_edge, end_edge);
+
+    double bisector_length = minimum_radius/std::sin(signed_angle/2);
+
+    circle_center_coordinate_.x = end_edge->start->coords.x + unit_bisector[0]*bisector_length;
+    circle_center_coordinate_.y = end_edge->start->coords.y + unit_bisector[1]*bisector_length;
+
+    minimum_radius_ = minimum_radius;
+    valid_corner_ = true;
+
+  }
+}
+
+void CornerArc::interpolateArc(const double & max_angle_resolution, std::vector<geometry_msgs::msg::PoseStamped> & poses)
+{
+
+  std::vector<double> r_start{ start_coordinate_.x-circle_center_coordinate_.x, start_coordinate_.y-circle_center_coordinate_.y };
+  std::vector<double> r_end{ end_coordinate_.x-circle_center_coordinate_.x,  end_coordinate_.y-circle_center_coordinate_.y };
+  double cross = r_start[0]*r_end[1] - r_start[1]*r_end[0];
+  double dot = r_start[0]*r_end[0] + r_start[1]*r_end[1];
+  signed_angle_ = std::atan2(cross, dot);
```

Comment:
I only see this being used in this function. I don't think it needs to be a class member